tests/Unit/Infrastructure/FileSyncer/RsyncFileSyncerTest.php

```diff
@@ -33,6 +33,8 @@ public function setUp(): void
         $this->filesystem
             ->exists(Argument::any())
             ->willReturn(true);
+        $this->filesystem
```

Comment:
Should this maybe have a `->shouldBeCalled()` prediction? Also, is there a more specific thing we could do than `Argument::any()`? (I'm thinking `Argument::type('string')`.)

Reply:
No, there are no predictions/expectations in the `::setup()` method. This is only here because of a quirk of Prophecy that if a test _happens_ to call a method on a mock that hasn't been declared as a possibility, it will fail with an unintuitive and very difficult-to-diagnose error. It's weird, but if I didn't do it here, I would have had to do it in seemingly random tests where it's not central to the purpose, obscuring the important expectations and leading to brittle tests. You'll see the same pattern all over the place in my tests. In fact, it's come up in code reviews before. 🙂